### DISCUSSION TITLE
create an unique interceptor instance

### DISCUSSION THREAD

**Discussion by smo043** [mswjs/mswjs.io#329] [Q&A] [answered]
I am looking for a method that works similarly to axios.create. How should I use this library to achieve a similar approach like Axios? Kindly assist.

**Comment by kettanaito** [comment, marked as answer]
Hey, @smo043. 

MSW works on an _endpoint basis_. Using [Request handlers](https://mswjs.io/docs/concepts/request-handler), you can describe the network you want on the request/response basis. 

```js
import { http, HttpResponse } from 'msw'

const handlers = [
  http.post('/login', () => HttpResponse.json({ success: true }))
]
```

**I highly recommend you go through our docs, they will teach you everything to [Get started with MSW](https://mswjs.io/docs/getting-started)**. 

If you are referring particularly to the encapsulation that `axios.create()` does, I believe the only thing transferrable from there to MSW is the base URL. MSW doesn't have a designated API for handling base URLs but you can do that easily yourself:

- https://github.com/mswjs/msw/issues/397#issuecomment-751230924

**Reply by smo043** [reply]
Thank you; I will give it a shot. We can close this discussion.
